Source/OxyPlot/Axes/Axis.cs

```diff
@@ -868,39 +869,74 @@ public virtual void Measure(IRenderContext rc)
                     throw new InvalidOperationException();
             }
 
-            var reachesMinPosition = Math.Min(this.StartPosition, this.EndPosition) < 0.01;
-            var reachesMaxPosition = Math.Max(this.StartPosition, this.EndPosition) > 0.99;
+            if (this.IsPanEnabled || this.IsZoomEnabled)
+            {
+                var reachesMinPosition = Math.Min(this.StartPosition, this.EndPosition) < 0.01;
+                var reachesMaxPosition = Math.Max(this.StartPosition, this.EndPosition) > 0.99;
 
-            switch (this.Position)
+                switch (this.Position)
+                {
+                    case AxisPosition.Left:
+                    case AxisPosition.Right:
+                        if (reachesMinPosition)
+                        {
+                            marginBottom = maximumTextSize.Height / 2;
+                        }
+
+                        if (reachesMaxPosition)
+                        {
+                            marginTop = maximumTextSize.Height / 2;
+                        }
+
+                        break;
+                    case AxisPosition.Top:
+                    case AxisPosition.Bottom:
+                        if (reachesMinPosition)
+                        {
+                            marginLeft = maximumTextSize.Width / 2;
+                        }
+
+                        if (reachesMaxPosition)
+                        {
+                            marginRight = maximumTextSize.Width / 2;
+                        }
+
+                        break;
+                }
+            }
+            else if (majorLabelValues.Count > 0)
             {
-                case AxisPosition.Left:
-                case AxisPosition.Right:
-                    if (reachesMinPosition)
-                    {
-                        marginBottom = maximumTextSize.Height / 2;
-                    }
+                var minLabel = majorLabelValues.Where(d => d > this.ActualMinimum).Min();
```

Comment:
If the `Where` filter is necessary, that would suggest that the filtering should be done once and cached, so that it can be confirmed that this list isn't empty (otherwise `Min`/`Max` will thrown).